### PR TITLE
Fix catch sdg null error on challenge

### PR DESCRIPTION
**Pull request by spuigmitjadxc**

### CODE REVIEW

File: app/views/decidim/challenges/challenges/show.html.erb [hunk 1]
```diff
@@ -37,30 +37,29 @@
...
-        </div>
+          <% if @challenge.sdg_code %>
+            <span class="definition-data__title"><%= t("sdg", scope: "activemodel.attributes.challenge") %></span>
+            <%= t_sdg(@challenge.sdg_code) %><br/>
+            <div class="ods">
+                <div class="text-container show">
+                    <p><%= t(@challenge.sdg_code + ".logo.line1", scope: "decidim.components.sdgs") %></p>
+                    <p><%= t(@challenge.sdg_code + ".logo.line2", scope: "decidim.components.sdgs") %></p>
+                </div>
+                <img alt="Logo ODS <%= (1 + Decidim::Sdgs::Sdg.index_from_code(@challenge.sdg_code.to_sym)).to_s.rjust(2, "0") %>"
```
Comment (Leusev) [inline app/views/decidim/challenges/challenges/show.html.erb:48]: Try to minimize business logic in the workforce please. 
You can solve this using some conditionals with the params that you may have prepared in the controller

File: app/controllers/decidim/challenges/challenges_controller.rb [hunk 2]
```diff
@@ -21,8 +21,6 @@ def index; end
 
       def show
         @challenge = Challenge.find(params[:id])
```
Comment (Leusev) [inline app/controllers/decidim/challenges/challenges_controller.rb:23]: Maybe here, you could initialize this deleted vars with some conditionals.
E.g.:
```
@sdg = @challenge.sdg_code if @challenge.sdg_code.present?
@sdg_index = (1 + Decidim::Sdgs::Sdg.index_from_code(@challenge.sdg_code.to_sym)).to_s.rjust(2, "0") if @sdg
```
or something like that...

**Review by Leusev — CHANGES_REQUESTED** [review]

**Comment by spuigmitjadxc** [comment]
@Leusev, all changes done

**Review by Leusev — APPROVED** [review]
Ok, looks good for me @spuigmitjadxc :+1:

**Pull request merged**